### DISCUSSION TITLE
Zoom with Trackpad vs Mousewheel

### DISCUSSION THREAD

**Discussion by CodeSippr** [clientIO/joint#2262] [Q&A] [answered]
### Introduction

Currently, I am using the mousewheel event to trigger a function that scales my dia.Paper in the background based on the delta of the event. This works fine with the mousewheel. However, when I use the pinch-to-zoom gesture on my trackpad, the scaling is too fast, and the entire page zooms in, causing the elements over the paper to be no longer visible. Is there any way to distinguish between the trackpad and the mousewheel using the jointjs events? Thank you for your assistance! :)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
Listen to the `paper:pinch` event instead.
Docs: https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.Paper.events

**Reply by CodeSippr** [reply]
The event looks perfect. It's just not triggered. Previously I had the following syntax in the paper:  
```
this.on("blank:mousewheel", (_evt, _x, _y, delta) => {
     console. log("blank:mousewheel", delta);
});
```

Now I tried it like this, but neither the pinch nor the pan event is triggered for me:  
```
this.on("paper:pinch", (evt: any, x: any, y: any, scale: any) => {
     console.log("pinch", scale);
});

this.on("paper:pan", (evt: any, deltaX: any, deltaY: any) => {
     console.log("pan", deltaX, deltaY);
});
```

**Reply by kumilingus** [reply, marked as answer]
What version do you use?
